modules/core/ui/theme/css/layout.css

```diff
@@ -1,10 +1,33 @@
-/* Add some vertical padding to layout regions. */
-.layout--twocol .layout__region {
-  margin-bottom: 1em;
+/* Add gap to layout. */
+.layout--twocol {
+  --layout-spacing: var(--gin-spacing-s);
+  gap: var(--layout-spacing);
+}
+
+/* Split first and second regions to two columns on large screens. */
+@media screen and (min-width: 40em) {
+  .layout--twocol > .layout__region--first,
+  .layout--twocol > .layout__region--second {
+    flex-basis: calc(50% - var(--layout-spacing)*.5);
+  }
 }
 
-/* Hide geometry field label in asset and log page top region. */
-body.path-asset .layout__region--top .field--name-geometry .field__label,
-body.path-log .layout__region--top .field--name-geometry .field__label {
+/* Hide layout regions if there is no content. */
+.layout--twocol .layout__region:not(:has(div)){
   display: none;
 }
```

Comment:
Hmm finding some situation where this might not work.

![Screenshot from 2024-04-23 12-53-30](https://github.com/farmOS/farmOS/assets/3116995/e028ad7b-a904-4363-974b-61a7aabc9845)

Reply:
This seemed to be caused by an empty `<div>` that seems like it would have displayed error/warning messages but it didn't render any and I haven't been able to reproduce. I was in-between switching branches and clearing cache so it's possible it was a fluke but I do wonder if this is an issue.